supervisor/api/backups.py

```diff
@@ -411,17 +419,29 @@ async def thaw(self, request: web.Request):
     async def remove(self, request: web.Request):
         """Remove a backup."""
         backup = self._extract_slug(request)
-        self._validate_cloud_backup_location(request, backup.location)
-        return self.sys_backups.remove(backup)
+        body = await api_validate(SCHEMA_REMOVE, request)
+        locations: list[LOCATION_TYPE] | None = None
+
+        if ATTR_LOCATION in body:
+            self._validate_cloud_backup_location(request, body[ATTR_LOCATION])
+            locations = [self._location_to_mount(name) for name in body[ATTR_LOCATION]]
+        else:
+            self._validate_cloud_backup_location(request, backup.location)
+
+        return self.sys_backups.remove(backup, locations=locations)
 
     @api_process
     async def download(self, request: web.Request):
         """Download a backup file."""
         backup = self._extract_slug(request)
-        self._validate_cloud_backup_location(request, backup.location)
+        # Query will give us '' for /backups, convert value to None
+        location = request.query.get(ATTR_LOCATION, backup.location) or None
+        self._validate_cloud_backup_location(request, location)
+        if location not in backup.all_locations:
+            raise APIError(f"Backup {backup.slug} is not in location {location}")
```

Comment:
Should that be an `APINotFound` error maybe? :thinking: I mean it is query string dependent, so not REST, maybe 404 is not appropriate then :thinking:

Reply:
Yea that was my thought process. The resource is the backup not the location so I left that as a 400. I could go either way though I suppose.